tests/scripts/all.sh

```diff
@@ -4385,14 +4385,14 @@ component_build_no_ssl_srv () {
     msg "build: full config except SSL server, make, gcc" # ~ 30s
     scripts/config.py full
     scripts/config.py unset MBEDTLS_SSL_SRV_C
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O1'
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O1 -Wmissing-prototypes'
 }
 
 component_build_no_ssl_cli () {
     msg "build: full config except SSL client, make, gcc" # ~ 30s
     scripts/config.py full
     scripts/config.py unset MBEDTLS_SSL_CLI_C
-    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O1'
+    make CC=gcc CFLAGS='-Werror -Wall -Wextra -O1 -Wmissing-prototypes'
```

Comment:
This [reveals another missing-prototype error](https://mbedtls.trustedfirmware.org/job/mbed-tls-pr-head/job/PR-9189-head/4/execution/node/2335/log/):
```
  CC    src/asn1_helpers.c
src/asn1_helpers.c:18:5: error: no previous prototype for 'mbedtls_test_asn1_skip_integer' [-Werror=missing-prototypes]
 int mbedtls_test_asn1_skip_integer(unsigned char **p, const unsigned char *end,
```

Please add the missing `#include`.

Reply:
Sorry, building that file allows the build to progress to another file with errors:

```
src/test_helpers/ssl_helpers.c:950:5: error: no previous prototype for ‘mbedtls_ssl_write_fragment’ [-Werror=missing-prototypes]
  950 | int mbedtls_ssl_write_fragment(mbedtls_ssl_context *ssl,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~
src/test_helpers/ssl_helpers.c:997:5: error: no previous prototype for ‘mbedtls_ssl_read_fragment’ [-Werror=missing-prototypes]
  997 | int mbedtls_ssl_read_fragment(mbedtls_ssl_context *ssl,
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~
```

You can test locally by running
```
tests/scripts/all.sh -k build_no_ssl_srv build_no_ssl_cli
```

Reply:
> Sorry, building that file allows the build to progress to another file with errors:
> 

Well, at least that's worth it 😂

> 
> You can test locally by running
> 
> ```
> tests/scripts/all.sh -k build_no_ssl_srv build_no_ssl_cli
> ```

Ah thanks! I was just asking that myself right away, so I could avoid flooding the CI :-)

Reply:
Okay, after installing a few missing Python plugins the scripts were working and this time the whole two prototypes were completely missing; At least a little variety twist ^^

But as a reward I received a few dozen more prototype errors:
```
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
```

I'll look at them later so don't hold your breath for now ;-)

Reply:
Turns out we unleashed prototype-hell ^^

Tuning `-ferror-limit=0` revealed 46+ prototype errors and consequently fixing them (and their follow-ups) just slightly inflated this PR.
Most of them were in `tests/suites` and I first considered implementing automatic `static`-prefix generation in `generate_test_code.py`. While browsing the `function` files  I noticed that there already was a mixture of static and non-static functions with different return types. So I went for the explicit way and changed all functions there. I think this is also the better approach for keeping the code clear (instead of having to guess what the generator script does).
However I also had to modify the script to generate static wrapper functions and identify existing static functions. Also refactored some stuff to satisfy the compiler (and the reader alike) in the sample programs where it felt appropriate.

Finally the only remaining error was triggered by the unused function `ecjpake_operation_setup` in `test_suite_psa_crypto.function` for which I took the liberty of commenting it out, since it is not referenced anywhere in the whole codebase.

If you agree with the recent modifications, do we need an additional changelog entry? And would you like me to add  `-ferror-limit=0` to `component_build_no_ssl_srv` and `component_build_no_ssl_cli` in `all.sh`?

Reply:
No, please don't add `-ferror-limit=0`. That's rarely useful: typically, when there are so many errors, it means the compiler is completely out of synch from what the programmer thought. We wouldn't want to generate GB worth of CI logs.